### PR TITLE
Update contributing guide

### PR DESCRIPTION
**Pull request by mayank99**
Fork instructions added. Updated links and checklist and some other stuff. Similar to https://github.com/iTwin/iTwinUI/pull/89

Preview at https://github.com/iTwin/iTwinUI-react/blob/mayank/update-contributing-guide/CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -125,34 +144,39 @@ Chokidar version:
 
 ### Changelog
 
-The `CHANGELOG.md` file must be updated for any new components or changes that you add. If unsure of which release your changes will go to, you can add a placeholder version and date (on the top):
+The `CHANGELOG.md` file must be updated for any new components or changes that you add. At the top of the file, create an "unreleased" section with a placeholder date (if not already present).
+
+<details>
```
Comment (bentleyvk) [inline CONTRIBUTING.md:149]: It is not that long, maybe no need to collapse.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,78 +2,91 @@
...
 `yarn createComponent`
 
 It ensures all needed imports are added and files are created.
 
-> Note: Every component needs to use `useTheme()` hook to ensure styling (theming) works fine. The script mentioned above adds it automatically.
-
-Be sure to develop on your branch. We want to keep branch naming with developer name scope:
+> Note: Every component needs to use `useTheme()` hook to ensure styling (theming) works fine. The `createComponent` script mentioned above adds it automatically.
 
-`git checkout -b yourName/your-feature-name`
```
Comment (veekeys) [inline CONTRIBUTING.md:40]: Sure we dont want to say anything about branch name?

Comment (mayank99) [reply CONTRIBUTING.md:40]: I think they can name their branch whatever they want. It's their fork 😄

File: CONTRIBUTING.md (hunk 1 above)
Comment (mayank99) [reply CONTRIBUTING.md:149]: I just kept the same as css repo. It helps with the readability of the whole document

**Pull request merged**